sonar-kotlin-checks/src/main/java/org/sonarsource/kotlin/checks/EqualsArgumentTypeCheck.kt

```diff
@@ -56,15 +60,19 @@ class EqualsArgumentTypeCheck : AbstractCheck() {
 
         val klass = function.containingClass() ?: return
         val parameter = function.valueParameters.first()
-        val parentNames = klass.superTypeListEntries.mapNotNull { it.typeReference!!.nameForReceiverLabel() }
+
 
         if (function.collectDescendantsOfType<KtIsExpression> { parameter.name == (it.leftHandSide as? KtNameReferenceExpression)?.getReferencedName() }
```

Comment:
I think this `if` is starting to become a little bit too messy and hard to read. Let's split the code a little bit so that it will be easier to change later.
GitHub does not allow me to make a proper suggestion on the whole method since some code was not changed, but I would replace it with something like:
```suggestion
                if (checkIsExpression(function, parameter, klass, bindingContext) &&
            checkWhenExpression(function, parameter, klass, bindingContext) &&
            checkBinaryExpression(function, parameter, klass) &&
            checkBinaryExpressionRHS(function, parameter, klass)
        ) {
            ctx.reportIssue(function.nameIdentifier!!, "Add a type test to this method.")
        }
```

Reply:
It is far better like that !!!